server/lib/notifications/agents/telegram.ts

```diff
@@ -134,6 +134,19 @@ class TelegramAgent
         disable_notification: this.getSettings().options.sendSilently,
       } as TelegramPayload);
 
+      if (
+        payload.notifyUser.settings?.enableNotifications &&
+        payload.notifyUser.settings?.telegramChatId
```

Comment:
What about a check here to see if the `telegramChatId` matches the global chat Id? To prevent duplicate notifications.